x/kavamint/abci.go

```diff
@@ -11,9 +11,12 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
 	if !found {
 		previousBlockTime = ctx.BlockTime()
 	}
+	// calculate totals before any minting is done to prevent new mints affecting the values
+	totalBonded := k.TotalBondedTokens(ctx)
+	totalSupply := k.TotalSupply(ctx)
 
 	// ------------- Staking Rewards -------------
-	stakingRewardCoins, err := k.AccumulateStakingRewards(ctx, previousBlockTime)
+	stakingRewardCoins, err := k.AccumulateStakingRewards(ctx, totalBonded, previousBlockTime)
```

Comment:
an orthogonal concern to this pr, but I kinda think it makes since for the first begin blocker to calculate the time since the last block and pass that to all modules so that is consistent across modules

Reply:
(perhaps by setting it as value in the context...)

Reply:
i've refactored to consolidate `AccumulateStakingRewards` and `AccumulateCommunityPoolInflation` into a single `AccumulateInflation(ctx, rate, basis, secondsSinceLastMint)` method.

Now `secondsPassed` is calculated in the begin blocker and the individual inflation calculations look like:
```go
	stakingRewardCoins, err := k.AccumulateInflation(
		ctx, params.StakingRewardsApy, totalBonded, secondsPassed,
	)

	communityPoolInflation, err := k.AccumulateInflation(
		ctx, params.CommunityPoolInflation, totalSupply, secondsPassed,
	)
```